cli/tests/pcluster/utils/test_pcluster_utils.py

```diff
@@ -189,14 +204,142 @@ def test_get_stack_resources(boto3_stubber, resources, error_message):
         )
     mocked_requests = [
         MockedBoto3Request(
-            method="describe_stack_resources", response=response, expected_params={"StackName": FAKE_STACK_NAME}
+            method="describe_stack_resources",
+            response=response,
+            expected_params={"StackName": FAKE_STACK_NAME},
+            generate_error=error_message is not None,
         )
     ]
-    generate_errors = error_message is not None
-    boto3_stubber("cloudformation", mocked_requests, generate_errors=generate_errors)
+    boto3_stubber("cloudformation", mocked_requests)
     if error_message is None:
         assert_that(utils.get_stack_resources(FAKE_STACK_NAME)).is_equal_to(resources)
     else:
         with pytest.raises(SystemExit, match=response) as sysexit:
             utils.get_stack_resources(FAKE_STACK_NAME)
         assert_that(sysexit.value.code).is_not_equal_to(0)
+
+
+def test_retry_on_boto3_throttling(boto3_stubber, mocker):
+    sleep_mock = mocker.patch("pcluster.utils.time.sleep")
+    mocked_requests = [
+        MockedBoto3Request(
+            method="describe_stack_resources",
+            response="Error",
+            expected_params={"StackName": FAKE_STACK_NAME},
+            generate_error=True,
+            error_code="Throttling",
+        ),
+        MockedBoto3Request(
+            method="describe_stack_resources",
+            response="Error",
+            expected_params={"StackName": FAKE_STACK_NAME},
+            generate_error=True,
+            error_code="Throttling",
+        ),
+        MockedBoto3Request(
+            method="describe_stack_resources", response={}, expected_params={"StackName": FAKE_STACK_NAME}
+        ),
+    ]
+    client = boto3_stubber("cloudformation", mocked_requests)
+    utils.retry_on_boto3_throttling(client.describe_stack_resources, StackName=FAKE_STACK_NAME)
+    sleep_mock.assert_called_with(5)
```

Comment:
I'd also add a check for the number of call, to be sure it reflects the number of throttling failures.
Something like: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_count

Reply:
tried to add this but other libraries are also calling the time.sleep function. We'll rely on the boto3 stubber to guarantee that the number of calls is correct